### PR TITLE
Temporary task completion system

### PR DESCRIPTION
**Pull request by zixm**
This PR included the following implementations:

    Task related Entities (Models)
    Task related Repositories
    Task Manager (Use Case)
    Task related Controllers
    Unit Testing (Github Actions Compatible)
    Java Documentation (JavaDoc)

Styling & Organization

    camelCasing Name Convention
    Mongo Repository compatible annotations (for clean data storage and retrieval)
    Commenting and JavaDoc where explanation non-trivial

Clean & SOLID
To the best of my ability, I have made a significant effort to adhere to SOLID principles, code smells and clean architecture. For example, Chain of Responsibility to decide whether or not the SessionID authorization is successful or not.

Testing
The coverage of this PR covers most basis with the test provided except for Task Manager.

Functionality
With the provided PR, I intend the group to be able to utilize the Task System (Task Completion Record), as well as setting up global Task Lists (Task) and Active Task Lists (Task Active).

JavaDoc
The coverage of this topic includes all methods and most classes where applicable (Entities, Controllers, Use Cases)

### CODE REVIEW

**Comment by rethegreat** [comment]
Code review:

This PR code review included the the review following implementations:

Task related Entities (Models)
Task related Repositories
Task Manager (Use Case)
Task related Controllers
Unit Testing (Github Actions Compatible)
Java Documentation (JavaDoc)
Styling & Organization

Clean & SOLID
From my observation this pr request adheres to SOLID principles, code smells and clean architecture. For example, Chain of Responsibility to decide whether or not the SessionID authorization is successful or not.

Testing
For testing postCompletedTaskTest is commented out, thus there is no longer a test for that necessary component. Thus also violating a code smell of unnecessary comments 
Along with a lot of testing are empty and needs to be implemented

Functionality
With the provided PR from my review it does utilize the Task System (Task Completion Record), as well as setting up global Task Lists (Task) and Active Task Lists (Task Active).


JavaDoc
The coverage of this topic includes all methods and most classes where applicable (Entities, Controllers, Use Cases)

**Comment by zixm** [comment]
Fixed conflicting files

**Comment by kurtislaw** [comment]
@rethegreat has done a comprehensive review and I agree with all his points. LGTM!

File: src/main/java/com/backend/entities/TaskCompletionRecord.java [hunk 1]
```diff
@@ -1,27 +1,93 @@
...
-    private final Timestamp timestamp;
-    private final String taskName;
+    //instance variables
+    @SuppressWarnings("unused")
+    @Id
+    private String ID;
+    @Transient
+    private final AccountID accountIDObject;
+    private final String accountID;
+    private final String timestamp;
```
Comment (TonyKim02) [inline src/main/java/com/backend/entities/TaskCompletionRecord.java:22]: change this from java.sql.timestamp to java.util.Date

Comment (zixm) [reply src/main/java/com/backend/entities/TaskCompletionRecord.java:22]: It's a String

**Comment by ShaanPurewal** [comment]
Remove PR (Outdated)

**Pull request closed without merging**